### ISSUE TITLE
SDR concatenate problem?

### ISSUE THREAD

**Issue by Thanh-Binh** [htm-community/htm.core#579]
I have compiling error in Ubuntu by using a vector for concatenating 2 SDRs together.
Can someone help me?
here is my code:
```
    SDR A({10});
    SDR B({10});
    SDR C({20});        
    SDR D({20});
    A.randomize( 0.30f );
    B.randomize( 0.70f );
    
    D.concatenate({&A, &B});	// OK

    std::vector<SDR*> AB;
    AB.push_back(&A);
    AB.push_back(&B);
    C.concatenate(AB);	// why failed???
```

**Comment by breznak** [comment]
hi @Thanh-Binh , thanks for testing the code! :+1: 
I don't see it right away, the compiler err would be helpful

**Comment by breznak** [comment]
ok, I got the error, it's in the const signature, which gets correct from the implicit `{&A, &B}` vector, but not from the explicit.

**Comment by breznak** [comment]
TL;DR: you've created vector of "incorrect" type. 
Should have been 
>  std::vector<const SDR*> AB;

or ideallly use the simplest `{&A, &B, ...&C }`

**Comment by Thanh-Binh** [comment]
@breznak Yes I see, best thanks.
